### ISSUE TITLE
Update JS dependencies

### ISSUE THREAD

**Issue by macfarla** [ConsenSys/permissioning-smart-contracts#232]
Many packages are out of date. Need to update lots of things. 
- [x] web3
- [x] ethers
- [x] jest
- [x] mocha-junit
- [x] replace Rimble UI -> Material UI
- [x] husky
- [x] prettier
- [x] typechain
These libraries can't be updated right now:
- react & react-dom
- truffle
- typescript

**Comment by macfarla** [comment]
https://github.com/ConsenSys/permissioning-smart-contracts/pull/231

**Comment by macfarla** [comment]
https://github.com/ConsenSys/permissioning-smart-contracts/pull/233

**Comment by macfarla** [comment]
updating to react 17 breaks the snapshots. There's no enzyme adapter yet for React 17 but they are working on it. https://github.com/enzymejs/enzyme/issues/2429

**Comment by macfarla** [comment]
https://github.com/ConsenSys/permissioning-smart-contracts/pull/234

**Comment by macfarla** [comment]
https://github.com/ConsenSys/permissioning-smart-contracts/pull/237
https://github.com/ConsenSys/permissioning-smart-contracts/pull/240

**Comment by macfarla** [comment]
typescript - `yarn run build` already gives this warning

**Comment by macfarla** [comment]
Husky https://github.com/ConsenSys/permissioning-smart-contracts/pull/241

**Comment by macfarla** [comment]
Truffle - tried updating to 5.4.7/8 but get a flaky test situation https://github.com/ConsenSys/permissioning-smart-contracts/pull/236

```
 Could not connect to your Ethereum client. Please check that your Ethereum client:     - is running     - is accepting RPC connections (i.e., "--rpc" or "--http" option is used in geth)     - is accessible over the network     - is properly configured in your Truffle configuration file (truffle-config.js) 
ProviderError: 
    at Context.it (test/test-node-rules-list.js:89:42)
```

**Comment by macfarla** [comment]
updated truffle to the latest 5.0 version - https://github.com/ConsenSys/permissioning-smart-contracts/pull/242

**Comment by macfarla** [comment]
react-scripts - updating from 3.0.1 to 3.1.0 gives this error on `yarn run build` (and following the given instructions did not help)
```

There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.

The react-scripts package provided by Create React App requires a dependency:

  "babel-eslint": "10.0.2"

Don't try to install it manually: your package manager does it automatically.
However, a different version of babel-eslint was detected higher up in the tree:

  /Users/sallymacfarlane/node_modules/babel-eslint (version: 10.0.1)

Manually installing incompatible versions is known to cause hard-to-debug issues.

If you would prefer to ignore this check, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That will permanently disable this message but you might encounter other issues.

To fix the dependency tree, try following the steps below in the exact order:

  1. Delete package-lock.json (not package.json!) and/or yarn.lock in your project folder.
  2. Delete node_modules in your project folder.
  3. Remove "babel-eslint" from dependencies and/or devDependencies in the package.json file in your project folder.
  4. Run npm install or yarn, depending on the package manager you use.
```

**Comment by macfarla** [comment]
Solium -> Ethlint https://github.com/ConsenSys/permissioning-smart-contracts/pull/243

**Comment by macfarla** [comment]
Released 2.1.0 https://github.com/ConsenSys/permissioning-smart-contracts/releases 
https://www.npmjs.com/package/@consensys-software/permissioning-smart-contracts
